src/autoDeleteMessages.ts

```diff
@@ -1,23 +1,54 @@
 import { BotContext } from './context';
 
-export interface AutoDeleteMessage {
-  [messageId: number]: number;
-}
+const expired = (createdAt: number, timeout: number) => {
+  const minute = 60000;
+  return Math.floor((Date.now() - createdAt) / minute) >= timeout;
+};
 
 export const runMessageRecycling = async (
   ctx: BotContext
 ): Promise<void> => {
-  const database = ctx.database;
+  const database = await ctx.loadDatabase();
```

Comment:
and this is how we fixed the issue with the database always referring to an outdated state

Reply:
Makes sense, instead of receiving an instance it receives a function that will return a fresh one every time this function runs :clap: